### PR TITLE
test(model): assert Model.bulkSave(...) works if some document is not modified

### PR DESCRIPTION
**Pull request by AbdelrahmanHafez**
re #10437

### CODE REVIEW

**Review by vkarpov15 — APPROVED** [review]
Thanks :+1:

File: test/model.test.js [hunk 1]
```diff
@@ -7557,6 +7557,20 @@ describe('Model', function() {
...
+        name: String
+      });
+
+      const User = db.model('User', userSchema);
+
+      return co(function*() {
+        const user = yield User.create({ name: 'Hafez' });
+
+        const res = yield User.bulkSave([user]);
+        assert.ok(res.result.ok);
```
Comment (vkarpov15) [inline test/model.test.js:7571]: I'd prefer if we didn't assert on particular properties in `res.result`. I'd just make this check that no error was thrown using `const err = yield User.bulkSave([user]).then(() => null, err => err); assert.ifError(err);`

Comment (AbdelrahmanHafez) [reply test/model.test.js:7571]: I agree, that's the way to go.

Comment (AbdelrahmanHafez) [reply test/model.test.js:7571]: I agree, that's the way to go.

**Comment by AbdelrahmanHafez** [comment]
Happy to help 👍

**Comment by Kamikadze4GAME** [comment]
Maybe to change this lines (`L3731-L3756`)

https://github.com/AbdelrahmanHafez/mongoose/blob/gh-10437/lib/model.js#L3731-L3756

into 

```js
    const isANewDocument = document.isNew;

    if (isANewDocument) {
      accumulator.push({
        insertOne: { document }
      });

      return accumulator;
    }

    const delta = document.$__delta();
    const isDocumentWithChanges = delta != null && !utils.isEmptyObject(delta[0]);

    if (isDocumentWithChanges) {
      const where = document.$__where(delta[0]);
      const changes = delta[1];

      _applyCustomWhere(document, where);

      document.$__version(where, delta);

      accumulator.push({
        updateOne: {
          filter: where,
          update: changes
        }
      });
      
      return accumulator;
    }

    return accumulator;
```

**Comment by AbdelrahmanHafez** [comment]
@Kamikadze4GAME Done 👍

**Review by Kamikadze4GAME — APPROVED** [review]

**Review by vkarpov15 — APPROVED** [review]

**Pull request merged**